app/src/main/java/com/example/moviesdatabase/MainActivity.java

```diff
@@ -8,22 +8,21 @@
 import android.widget.Toast;
 
 public class MainActivity extends AppCompatActivity {
-
+    Button retry;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
-        Button retry = findViewById(R.id.retry);
-        retry.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Toast.makeText(getApplicationContext(), getString(R.string.toast_retry), Toast.LENGTH_SHORT).show();
-
-            }
-        });
+        retry = findViewById(R.id.retryButt);
     }
 
-
-
-}
\ No newline at end of file
+   public void checker(View view) {
+        /*
+        InternetConnectionChecker checker = new InternetConnectionChecker();*/
+        if (InternetConnectionChecker.hasConnection(getApplicationContext())){
```

Comment:
to avoid code repetition you could do smth like the following: 
```

        int toastMessageResId;
        if (InternetConnectionChecker.hasConnection(getApplicationContext())) {
            toastMessageResId = R.string.toast_retry;
        } else {
            toastMessageResId = R.string.toast_please_connect;
        }
        Toast.makeText(getApplicationContext(), getString(toastMessageResId), Toast.LENGTH_SHORT).show();
```